src/_queries/useGetAsyncStorageValue.ts

```diff
@@ -0,0 +1,16 @@
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import { useQuery } from '@tanstack/react-query';
+import { StorageKey } from '../_models';
+
+async function getAsyncStorageValue<T>(storageKey: StorageKey): Promise<T> {
+  const storageValue = await AsyncStorage.getItem(storageKey);
+  try {
+    return JSON.parse(storageValue) as T;
+  } catch (error) {
+    return storageValue as any;
```

Comment:
Why would you cast this to `any`? 🤔

Reply:
omdat we niet weten wat dat kan zijn dat, misschien wel een use case al voor zod 🤔